### ISSUE TITLE
Audible clicks with some video files

### ISSUE THREAD

**Issue by dos1** [opensludge/opensludge#16]
I've tried looking into why playing https://dosowisko.net/sludge/dosowisko30.webm with SLUDGE causes audible clicks.

Turns out that clicks happen in places where `vorbis_synthesis_pcmout` returns a lower value than usual. Usually it returns 4096, for other files I tested even all the time, however, in this file it sometimes returns lower values: 2304 and 512. The moments it happens correlate with audible clicks.

So, I have looked into how the audio samples are queued and fed into ALURE.

https://github.com/opensludge/opensludge/blob/df0989e96d71a1578c55fb97a673d53907346c68/source/Engine/movie.cpp#L293-L339

The code looks like it used to be a loop. When wrapping most of it with `while (length > 0)` and modifying the `memcpy` call a bit to take previous loop rolls into account (`memcpy(data+got, buffer+bufOffset, bufLen);`), the clicks are fixed - however, the A/V sync is now broken (the video is way too fast).

However, when adding `if (length > 4096) { length = 4096; }` before the loop, audible clicks are gone and A/V sync is left intact. This is a very ugly hack though and I'm not sure yet how to deal with it properly, so I'm opening this ticket instead.

Any ideas?

PS. Those two lines should be swapped around, shouldn't they?
https://github.com/opensludge/opensludge/blob/df0989e96d71a1578c55fb97a673d53907346c68/source/Engine/movie.cpp#L332-L335

**Comment by tobihan** [comment]
Yes, I'm afraid movie support was never tested that much and might need some improvement.

> PS. Those two lines should be swapped around, shouldn't they?

Yes.
